inspirehep/search/factories/search.py

```diff
@@ -27,11 +30,9 @@ def inspire_search_factory(self, search):
 
     try:
         search = search.query_from_iq(query_string)
-    except SyntaxError:
-        current_app.logger.debug(
-            "Failed parsing query: '%s'", request.values.get("q", ""), exc_info=True
-        )
-        raise InvalidQueryRESTError()
+    except SyntaxError as exc:
+        LOGGER.warning("Failed parsing query: '%s'", request.values.get("q", ""))
+        raise InvalidQueryRESTError() from exc
```

Comment:
Like this we are loosing the SyntaxError exception, no? Do we want to shallow the SyntaxError?

Reply:
We are not loosing it. It will be still displayed in stack trace:
`Exception                                 Traceback (most recent call last)
<ipython-input-1-c8abcbc6288a> in <module>
      1 try:
----> 2     raise Exception('aaaa')
      3 except Exception as e:

Exception: aaaa

The above exception was the direct cause of the following exception:

TimeoutError                              Traceback (most recent call last)
<ipython-input-1-c8abcbc6288a> in <module>
      2     raise Exception('aaaa')
      3 except Exception as e:
----> 4     raise TimeoutError() from e
      5 

TimeoutError: 
`